### PR TITLE
Allow shape and chunks to be integers when creating dataset

### LINKED ISSUE #1229: Accept integers for 1D shapes
I am trying to manually set the chunking options in a 1D array. But the obvious ways to do this fail.

```python
import h5py
import numpy as np

data = np.random.randint(0, 1000, 1000000, dtype = "i2")

f = h5py.File("array_data.hdf5", "w")
f.create_dataset("data", data = data, dtype = "i2", chunks = 50000)
f.close()
```

This gives error:

```python
>>> f.create_dataset("data", data = data, dtype = "i2", chunks = 50000)
Traceback (most recent call last):
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 129, in rq_tuple
    tpl = tuple(tpl)
TypeError: 'int' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/group.py", line 136, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/dataset.py", line 140, in make_new_dset
    maxshape, scaleoffset, external)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 135, in fill_dcpl
    rq_tuple(chunks, 'chunks')
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 131, in rq_tuple
    raise TypeError('"%s" argument must be None or a sequence object' % name)
TypeError: "chunks" argument must be None or a sequence object
```

Alternatively making the chunks a tuple of 1

```python
f = h5py.File("array_data.hdf5", "w")
f.create_dataset("data", data = data, dtype = "i2", chunks = (50000))
f.close()
```

fails due to 

```python
>>> f.create_dataset("data", data = data, dtype = "i2", chunks = (50000))
Traceback (most recent call last):
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 129, in rq_tuple
    tpl = tuple(tpl)
TypeError: 'int' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/group.py", line 136, in create_dataset
    dsid = dataset.make_new_dset(self, shape, dtype, data, **kwds)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/dataset.py", line 140, in make_new_dset
    maxshape, scaleoffset, external)
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 135, in fill_dcpl
    rq_tuple(chunks, 'chunks')
  File "/anaconda/envs/long-read-methylation/lib/python3.7/site-packages/h5py/_hl/filters.py", line 131, in rq_tuple
    raise TypeError('"%s" argument must be None or a sequence object' % name)
TypeError: "chunks" argument must be None or a sequence object
```

So is it possible to set chunks for 1D arrays?

 * Operating System : MacOS 10.11
 * Python version: 3.7.3
 * Python acquired: Anaconda on MacOS
 * h5py version: 2.9.0
 * HDF5 version: 1.10.4

### PR DESCRIPTION
**Pull request by juliagarriga**
Related to #1229

### CODE REVIEW

**Comment by takluyver** [comment]
Thanks Julia, this looks good to me. Other maintainers: any objections to allowing an integer anywhere we accept a shape tuple?

There are a few other shape-like parameters where I think the same pattern should apply:

- [ ] The `maxshape` parameter in the code path where you already worked on `shape` and `chunks` (for `grp.create_dataset`)
- [ ] `grp.require_dataset(shape=...)`
- [x] `grp.create_dataset_like()` - the changes you already made should cover this
- [ ] `attrs.create(shape=...)`
- [ ] `VirtualSource(shape=)`
- [ ] `VirtualSource(maxshape=)`
- [ ] `VirtualLayout(shape=)`
- [ ] `VirtualLayout(maxshape=)`

**Comment by scopatz** [comment]
Thanks @juliagarriga! Please add a news entry, as per the instructions here: https://regro.github.io/rever-docs/news

**Comment by takluyver** [comment]
Thanks. Can you see where to make changes for the other shape parameters I listed above?

I think the changes are small enough that it makes sense to do them all in one PR, but if it's not clear what to do, or you don't have time, we can leave the others for later PRs.

File: h5py/_hl/dataset.py [hunk 1]
```diff
@@ -90,12 +90,17 @@ def make_new_dset(parent, shape=None, dtype=None, data=None,
             data = Empty(dtype)
         shape = data.shape
     else:
-        shape = tuple(shape)
+        shape = tuple((shape,)) if isinstance(shape, int) else tuple(shape)
```
Comment (takluyver) [inline h5py/_hl/dataset.py]: I think the first `tuple()` here is unnecessary: `(shape,)` makes a tuple already.

**Comment by juliagarriga** [comment]
> There are some simple tests of VirtualSource here: https://github.com/h5py/h5py/blob/master/h5py/tests/test_vds/test_virtual_source.py

Yes for VirtualSource I added the tests: test_integer_shape, test_integer_maxshape

**Comment by takluyver** [comment]
Ah, sorry, I missed that you'd already added those. This looks good to me, then.

**Comment by takluyver** [comment]
I think the 'closes #1229' needs to be in the commit message for it to close the issue automatically on merge. I've closed #1229 manually now. :-)

**Comment by juliagarriga** [comment]
@takluyver Ok perfect! :+1:

**Pull request merged**